### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Lucas-Masaba**
# Project Requirements ⚙️ 
- Create models
- Use models to insert data
- Create custom methods

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (hiromataba) [inline app/models/user.rb]: - [x] Please add a foreign key for posts, likes, and comments to set up the associations 

```ruby
has_many :posts, foreign_key: 'user_id'
has_many :comments, foreign_key: 'user_id'
has_many :likes, foreign_key: 'user_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :comments
+  belongs_to :user, foreign_key: :user_id
+
+  def recent_comments
+    comments.last(5)
+  end
+
+  def update_post_user_counter
+    user.update(posts_counter: user.posts.count)
+  end
```
Comment (hiromataba) [inline app/models/post.rb:16]: - [x] [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_user_counter` method every time a new post has been saved to the database, then make the `update_post_user_counter` method private

`after_save :update_post_user_counter`

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @Lucas-Masaba     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Very professional README file :+1: 
- Nice commit history :heavy_check_mark: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Lucas-Masaba) [reply app/models/user.rb]: Thank you for the suggestion. I have successfully implemented the changes

File: app/models/post.rb (hunk 2 above)
Comment (Lucas-Masaba) [reply app/models/post.rb:16]: I read about it and I found it useful. Thank you.

**Review by DeliceLydia — APPROVED** [review]
## Hi @Lucas-Masaba, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**